homeassistant/components/sleepiq/number.py

```diff
@@ -101,6 +104,27 @@ def _get_foot_warming_unique_id(bed: SleepIQBed, foot_warmer: SleepIQFootWarmer)
     return f"{bed.id}_{foot_warmer.side.value}_{FOOT_WARMING_TIMER}"
 
 
+async def _async_set_core_climate_time(
+    core_climate: SleepIQCoreClimate, time: int
+) -> None:
+    temperature = CoreTemps(core_climate.temperature)
+    if temperature != CoreTemps.OFF:
+        await core_climate.turn_on(temperature, time)
+
+    core_climate.timer = time
+
+
+def _get_core_climate_name(bed: SleepIQBed, core_climate: SleepIQCoreClimate) -> str:
+    sleeper = sleeper_for_side(bed, core_climate.side)
+    return f"SleepNumber {bed.name} {sleeper.name} {ENTITY_TYPES[CORE_CLIMATE_TIMER]}"
```

Comment:
Instead of this, please look into entity name translation where the bed name and sleeper name can be injected as place holders.
Also, why does the name start with "SleepNumber"?

Reply:
Please add more context into what you're requesting. 

This pattern is what is used throughout this integration that has been previously approved and currently exists within HA. See lines 63, 80, 98. 

I do not believe we should be changing this as a one off within the integration. 

I would like to further understand what you're requesting. If you're suggesting this should be changed through the entire integration for this PR to be approved, @Kbickar would need to chime in. As I am not the owner of the integration.

Reply:
SleepNumber is the name of the company/bed this integration supports. 

https://www.sleepnumber.com/

Reply:
The original integration was from a very Early version of HA and contained that prefix for the sensors. In order to keep backwards compatibility and consistency , we kept the prefix for all of them